server/src/main/java/org/elasticsearch/index/mapper/TextFieldMapper.java

```diff
@@ -1008,17 +1008,21 @@ protected String delegatingTo() {
                     }
                 };
             }
+            if (isStored()) {
+                return new BlockStoredFieldsReader.BytesFromStringsBlockLoader(name());
+            }
             /*
              * If this is a sub-text field try and return the parent's loader. Text
              * fields will always be slow to load and if the parent is exact then we
              * should use that instead.
              */
+            // TODO: should this be removed? I think SyntheticSourceHelper already does this:
```

Comment:
`SyntheticSourceHelper` is concerned with multi fields of the `text` field. This logic is for when `text` field is the multii field of a `keyword` field.

```
{
  "t": {
    "type": "text",
    "fields": {
      "k": {
        "type": "keyword"
      }
  }
}
```
vs

```
{
  "k": {
    "type": "keyword",
    "fields": {
      "t": {
        "type": "text"
      }
    }
  }
}
```